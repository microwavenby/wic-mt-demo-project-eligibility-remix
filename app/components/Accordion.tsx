import { Trans } from "react-i18next";
import { ReactElement } from "react";
import { Accordion as USWDS_Accordion } from "@trussworks/react-uswds";
import { i18nKey } from "app/types";

export type AccordionProps = {
  bodyKey: i18nKey;
  headerKey: i18nKey;
  id: string;
};

const Accordion = (props: AccordionProps): ReactElement => {
  const { bodyKey, headerKey, id } = props;

  return (
    <USWDS_Accordion
      bordered={true}
      items={[
        {
          title: <Trans i18nKey={headerKey} />,
          content: <Trans i18nKey={bodyKey} />,
          id: id,
          expanded: false,
          headingLevel: "h3",
        },
      ]}
    />
  );
};

export default Accordion;
