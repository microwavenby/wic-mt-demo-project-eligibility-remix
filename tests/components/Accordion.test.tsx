import { render, screen } from "@testing-library/react";

import Accordion from "app/components/Accordion";

import { setupUserEvent } from "../helpers/setup";

const testProps = {
  bodyKey: "body",
  headerKey: "header",
  id: "test-accordion",
};

it("should match snapshot", () => {
  const { container } = render(<Accordion {...testProps} />);
  expect(container).toMatchSnapshot();
});

it("should expand the accordion on click", async () => {
  const user = setupUserEvent();
  render(<Accordion {...testProps} />);

  const body = screen.getByText(testProps.bodyKey);
  expect(body.hidden).toBe(true);

  const button = screen.getByText(testProps.headerKey);
  await user.click(button);
  expect(body.hidden).toBe(false);
});
