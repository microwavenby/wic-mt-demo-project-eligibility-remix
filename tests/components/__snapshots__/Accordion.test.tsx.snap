// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="usa-accordion usa-accordion--bordered"
    data-testid="accordion"
  >
    <h3
      class="usa-accordion__heading"
    >
      <button
        aria-controls="test-accordion"
        aria-expanded="false"
        class="usa-accordion__button"
        data-testid="accordionButton_test-accordion"
        type="button"
      >
        header
      </button>
    </h3>
    <div
      class="usa-accordion__content usa-prose"
      data-testid="accordionItem_test-accordion"
      hidden=""
      id="test-accordion"
    >
      body
    </div>
  </div>
</div>
`;
